gulpfile.js

```diff
@@ -2,13 +2,15 @@ const gulp = require('gulp')
 const del = require('del')
 const shell = require('gulp-shell')
 const ts = require('gulp-typescript')
+const typedoc = require('gulp-typedoc')
 const mocha = require('gulp-mocha')
 const tslint = require('gulp-tslint')
 const distFiles = require('./package.json').files
 
-const PRIVATE_INTERFACE_FILES = ["./src/lib/**/*.d.ts"]
 const SOURCE_FILES = ["./src/**/*.ts", "./src/**/*.tsx"]
```

Comment:
src/**/*.ts?(x)

Is (if this is a regex?) equivalent to duplicating between checking ts and not ts

Also, on other projects I've had some files missed using this. I'm unsure if ./src/**/*.ts matches a file in src E.g.

/src/hello.ts - would this be caught by this? I know on other projects I've had to do 'src/*.ts?(x)' 'src/**/*.ts?(x)' to get it to work..

Reply:
With gulp, `**` in a path pattern will match same directory, so including "./src/lib/*.d.ts" isn't necessary. I had to confirm that though, so good question!